doc/crypto/api/ops/key-encapsulation.rst

```diff
@@ -0,0 +1,350 @@
+
+    A call to `psa_encapsulate()` carries out steps 1 to 4 of the ECIES encryption process described in `[SEC1]` §5.1.3:
+
+    *   The elliptic curve to use is determined by the key.
+    *   The public key part of the input key is used as :math:`Q_V`.
+    *   Cofactor ECDH is used to perform the key agreement.
+    *   The shared secret :math:`Z` is output as the shared output key.
+    *   The ephemeral public key :math:`\overline{R}` is output as the encapsulation data.
+
+    A call to `psa_decapsulate()` carries out steps 2 to 5 of the ECIES decryption process described in `[SEC1]` §5.1.4:
+
+    *   The elliptic curve to use is determined by the key.
+    *   The encapsulation data is decoded as :math:`\overline{R}`.
+    *   The private key of the input key is used as :math:`d_V`.
+    *   Cofactor ECDH is used to perform the key agreement.
+    *   The shared secret :math:`Z` is output as the shared output key.
+
+    The encapsulation data produced by `PSA_ALG_ECIES_SEC1` is not authenticated.
+    When used in a full ECIES scheme, the authentication of the encrypted message implicitly confirms that the derived keys were identical.
+
+    The shared output key that is produced by `PSA_ALG_ECIES_SEC1` is not suitable for use as an encryption key.
+    It must be used as an input to a key derivation operation to produce additional cryptographic keys.
+
+    .. subsection:: Compatible key types
+
+        | :code:`PSA_KEY_TYPE_ECC_KEY_PAIR(family)`
+        | :code:`PSA_KEY_TYPE_ECC_PUBLIC_KEY(family)` (encapsulaton only)
+
+        where ``family`` is a Weierstrass or Montgomery Elliptic curve family.
+        That is, one of the following values:
+
+        *   ``PSA_ECC_FAMILY_SECT_XX``
+        *   ``PSA_ECC_FAMILY_SECP_XX``
+        *   `PSA_ECC_FAMILY_FRP`
+        *   `PSA_ECC_FAMILY_BRAINPOOL_P_R1`
+        *   `PSA_ECC_FAMILY_MONTGOMERY`
+
+Key-encapsulation functions
+---------------------------
+
+.. function:: psa_encapsulate
+
+    .. summary::
+        Use a public key to generate a new shared secret key and associated encapsulation data.
+
+    .. param:: psa_key_id_t key
+        Identifier of the key to use for the encapsulation.
+        It must be a public key or an asymmetric key pair.
+        It must permit the usage `PSA_KEY_USAGE_ENCAPSULATE`.
+    .. param:: psa_algorithm_t alg
+        The key-encapsulation algorithm to use: a value of type `psa_algorithm_t` such that :code:`PSA_ALG_IS_KEY_ENCAPSULATION(alg)` is true.
+    .. param:: const psa_key_attributes_t * attributes
+        The attributes for the output key.
+        This function uses the attributes as follows:
+
+        *   The key type.
+            All key-encapsulation algorithms can output a key of type :code:`PSA_KEY_TYPE_DERIVE` or :code:`PSA_KEY_TYPE_HMAC`.
+            Key-encapsulation algorithms that produce a pseudo-random shared secret, can also output block-cipher key types, for example :code:`PSA_KEY_TYPE_AES`.
```

Comment:
This wording was duplicated from the PAKE `psa_pake_get_shared_key()` API, which has similar properties, depending on the PAKE algorithm in use. Some PAKEs ensure that the shared secret is uniformly pseudorandom, and of a useful size for keying material. So we worded the API description to permit immediate use as a key to permit use cases where the higher level protocol does not do any further key derivation.

I don't think we can rule out such higher level protocol designs for encapsulation, given the following text from FIPS203 (ML-KEM):

> **Approved usage of the shared secret key.** If randomness generation is successful, the values *K* and *K'* returned by ML-KEM.Encaps and ML-KEM.Decaps, respectively, are always 256-bit values. Under appropriate conditions (see Sections 3.1 and 3.2, and SP 800-227), these values match (i.e., *K = K'*) and can be used directly as a shared secret key for symmetric cryptography. If further key derivation is needed, the final symmetric keys shall be derived from this 256-bit shared secret key in an approved manner, as specified in SP 800-108 and SP 800-56C.

Reply:
There is already a statement to direct the reader to the individual key-encapsulation algorithm documentation for specific details.

Reply:
AFAIK it's not that common to directly use a PAKE output as a “proper” key, rather than a verification value or fodder for derivation. On the other hand, it's pretty common after a key agreement to want to do minimal work to use the shared secret as an AEAD key. That's only secure if the shared secret is uniformly pseudorandom (which is not the case for the dominant key agreement, namely FFDH/ECDH), and that tends to require truncation. For a KEM designed to have uniformly pseudorandom output, I expect protocols will specify things like “use the first 128 bits as the AES key”.